app/src/org/commcare/CommCareApplication.java

```diff
@@ -1243,4 +1245,20 @@ public void onStateChanged(@NonNull LifecycleOwner source, @NonNull Lifecycle.Ev
                 break;
         }
     }
+
+    public IDatabase createOrOpenGlobalDatabase() {
+        return new EncryptedDatabaseAdapter(new DatabaseGlobalOpenHelper(this, "null"));
+    }
+
+    public IDatabase createOrOpenUserDatabase(String userKeyRecordId, String key) {
+        return new EncryptedDatabaseAdapter(new DatabaseUserOpenHelper(this, userKeyRecordId, key));
+    }
+
+    public IDatabase openUserDatabase(String path, String password) {

```

Comment:
Curious if we have tested the pathway that leads to this method ?

Reply:
I don't have any context here as well, we probably need to trace the code and then ask related questions to HQ dev or Clayton on it.

Reply:
thanks, that seems like enough testing to me on that flow. Given that we can't let QA test it, think developers are accountable for testing such changes thoroughly and it should form a critical part of safety story on the PR.